### ISSUE TITLE
Failing to get light signed header at height

### ISSUE THREAD

**Issue by jtieri** [cosmos/relayer#697]
Looks like somehow we are trying to query the wrong chain's height in `(*CosmosProvider).InjectTrustedFields`
I only see this bug when running the relayer in production settings on paths that have higher traffic or more channels.

```
2022-04-09T20:37:12.246420Z     info    Failed to relay packet from sequence    {"src_chain_id": "osmosis-1", "src_channel_id": "channel-0", "src_port_id": "transfer", "dst_chain_id": "cosmoshub-4", "dst_channel_id": "channel-141", "dst_port_id": "transfer", "attempt": 3, "attempt_limit": 5, "error": "failed to get trusted header, please ensure header at the height 10058691 has not been pruned by the connected node: non-tm client header"}
```

For example, I can leave the relayer running on Sommelier<->Osmosis where there is only one channel and very little traffic and I won't see this issue. If I try running on Cosmos Hub<->Osmosis, where there are 5 channels and high traffic, the errors show up with some frequency over a longer period of time.

**Comment by tpkeeper** [comment]
got the same err

**Comment by jtieri** [comment]
Should be closed by #718
